crypto/ec/ecx_meth.c

```diff
@@ -966,7 +981,7 @@ static int s390x_x448_mul(unsigned char u_dst[56],
     memcpy(param.x448.d_src, d_src, 56);
 
     s390x_flip_endian64(param.x448.u_src, param.x448.u_src);
-    s390x_x448_mod_p(param.x448.u_src);
+    s390x_x448_mod_p(param.x448.u_src + 8);
```

Comment:
I'm struggling to understand what's going on in this change and the other related change in your first commit. Can you explain in a bit more detail what is happening here?

Reply:
The structure of the parameter block the pcc instruction works on is reflected by the param union. The 56 byte values x448 operates on are stored right-aligned in its 64 byte fields. The x448_mod_p reduction function works on 56 bytes, therefore it must start on an 8 bytes offset into the buffer. The bug was that without the offset, the leading 8 bytes were always zero, so reduction never kicked in.

Reply:
The second bug in the reduction (which was obv. shadowed by the first bug) was that i should have checked one bit higher for overflow at the end of the reduction (charck the least significant carry bit instead of most significant bit in most significant byte of u_red).